crates/oci-distribution/src/lib.rs

```diff
@@ -201,14 +183,18 @@ impl Challenge for BearerChallenge {
             }),
         }
     }
+
     fn into_raw(self) -> RawChallenge {
         let mut map = ChallengeFields::new();
-        self.realm
-            .and_then(|realm| map.insert_static_quoting("realm", realm));
-        self.scope
-            .and_then(|item| map.insert_static_quoting("scope", item));
-        self.service
-            .and_then(|item| map.insert_static_quoting("service", item));
+        if let Some(realm) = self.realm {
```

Comment:
Okay, I have to say that I find the if/map thing less readable than the and_then() thing. Is this an established pattern or a preference?

Reply:
I would say using `if let` is much more common than `and_then` in this case. `and_then` is really best when wanting to do the same as a `map` then `flatten`. We simply want to perform an action if the item is set.